### PR TITLE
feat(curriculum): add bisection method project to Python

### PR DESCRIPTION
**Pull request by zairahira**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [ ] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX
Adding a new project to the Python curriculum.

### CODE REVIEW

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1c34a541afc12dbb849d.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
+id: 65ef1c34a541afc12dbb849d
+title: Step 13
+challengeType: 20
+dashedName: step-13
+---
+
+# --description--
+
+Now, check if the absolute difference between `square_mid` and `square_target` is within the specified `tolerance`.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1c34a541afc12dbb849d.md:10]:
```suggestion
Now, create an `if` statement to check if the absolute value of the difference between `square_mid` and `square_target` is within the specified `tolerance`.
```
Before this I'd introduce `abs()`.

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1cda150a59c3b8306944.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+---
+id: 65ef1cda150a59c3b8306944
+title: Step 15
+challengeType: 20
+dashedName: step-15
+---
+
+# --description--
+
+If the difference is not within the specified `tolerance`, create and `elif` statement to check if `square_mid` is less than `square_target`.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1cda150a59c3b8306944.md:10]:
```suggestion
If the difference is not within the specified `tolerance`, create an `elif` statement to check if `square_mid` is less than `square_target`.
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d104c2be9c4d8e22075.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+```
+
+You should assign the value of `mid` to `high` in the `else` statement.
+
+```js
+({
+    test: () => 
+    {
+        assert(runPython(`_Node(_code).find_function("square_root_bisection").find_ifs()[1].find_bodies()[2].find_for_loops()[0].find_ifs()[0].find_bodies()[2]`));
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d104c2be9c4d8e22075.md:32]:
```suggestion
        assert(runPython(`_Node(_code).find_function("square_root_bisection").find_ifs()[1].find_bodies()[2].find_for_loops()[0].find_ifs()[0].find_bodies()[2].is_equivalent("high = mid")`));
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d104c2be9c4d8e22075.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+({
+    test: () => 
+    {
+        assert(runPython(`_Node(_code).find_function("square_root_bisection").find_ifs()[1].find_bodies()[2].find_for_loops()[0].find_ifs()[0].find_conditions()[2]== _Node()`));
+    }
+})
+
+```
+
+You should assign the value of `mid` to `high` in the `else` statement.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d104c2be9c4d8e22075.md:26]:
```suggestion
You should assign the value of `mid` to `high` in the `else` clause.
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d104c2be9c4d8e22075.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+dashedName: step-16
+---
+
+# --description--
+
+If both the `if` and `elif` conditions are not met, the value of `mid` would be greater than `square_target`. In this case, create an `else` statement and assign the value of `mid` to `high`.
+
+# --hints--
+
+You should have an `else` statement.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d104c2be9c4d8e22075.md:14]:
```suggestion
You should have an `else` clause.
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d104c2be9c4d8e22075.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
+---
+id: 65ef1d104c2be9c4d8e22075
+title: Step 16
+challengeType: 20
+dashedName: step-16
+---
+
+# --description--
+
+If both the `if` and `elif` conditions are not met, the value of `mid` would be greater than `square_target`. In this case, create an `else` statement and assign the value of `mid` to `high`.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d104c2be9c4d8e22075.md:10]:
```suggestion
If both the `if` and `elif` conditions are not met, the value of `mid` would be greater than `square_target`. In this case, create an `else` clause and assign the value of `mid` to `high`.
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d5e3d2927c5e0f4997b.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
+---
+id: 65ef1d5e3d2927c5e0f4997b
+title: Step 17
+challengeType: 20
+dashedName: step-17
+---
+
+# --description--
+
+Outside the for loop, create an `if` condition to check if the value of `root` is still `None`. If it is, print the message `'Failed to converge within {max_iterations} iterations.'`. Remember to format using an f-string.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d5e3d2927c5e0f4997b.md:10]:
```suggestion
Outside the for loop, create an `if` statement to check if the value of `root` is still `None`. If it is, print the message `'Failed to converge within {max_iterations} iterations.'`. Remember to format the message using an f-string.
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d5e3d2927c5e0f4997b.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+# --hints--
+
+You should create an `if` condition that checks `if root is None`.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d5e3d2927c5e0f4997b.md:14]:
```suggestion
You should create an `if` statement that checks if `root is None`.
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d5e3d2927c5e0f4997b.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+({
+    test: () => 
+    {
+        assert(runPython(`_Node(_code).find_function("square_root_bisection").find_ifs()[1].find_bodies()[2].find_ifs()[0].find_conditions()[0].is_equivalent("root is None")`));
+
+    }
+})
+```
+
+You should assign the parameter `f'Failed to converge within {max_iterations} iterations.'` to the `print` function.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d5e3d2927c5e0f4997b.md:26]:
```suggestion
You should pass the argument `f'Failed to converge within {max_iterations} iterations.'` to the `print` function.
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d924ff2e5c7031c6ce8.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
+---
+id: 65ef1d924ff2e5c7031c6ce8
+title: Step 18
+challengeType: 20
+dashedName: step-18
+---
+
+# --description--
+
+Create an `else` condition to handle the case where the value of `root` is not `None` and found as a result of convergance. If it is not `None`, print the message `'The approximate square root of {square_target} is {root}.'`. Remember to format using an f-string.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d924ff2e5c7031c6ce8.md:10]:
```suggestion
Create an `else` clause to handle the case where the value of `root` is not `None` and found as a result of convergence. If it is not `None`, print the message `'The square root of {square_target} is approximately {root}'`. Remember to format the message using an f-string.
```

Comment (Dario-DC) [reply curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d924ff2e5c7031c6ce8.md:10]: Maybe it's just me, but "found as a result of convergence" does not sound right.

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d924ff2e5c7031c6ce8.md [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+# --hints--
+
+There should be an `else` condition
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d924ff2e5c7031c6ce8.md:14]:
```suggestion
You should create an `else` clause.
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d924ff2e5c7031c6ce8.md [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+({
+    test: () => 
+    {
+         assert(runPython(`_Node(_code).find_function("square_root_bisection").find_ifs()[1].find_bodies()[2].find_ifs()[0].find_conditions()[1] == _Node()`));
+    }
+})
+
+```
+
+The body of `else` should print
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d924ff2e5c7031c6ce8.md:26]:
```suggestion
You should print `f'The square root of {square_target} is approximately {root}'` within the `else` body.
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1dd722f6e7c8294eeec4.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
+---
+id: 65ef1dd722f6e7c8294eeec4
+title: Step 19
+challengeType: 20
+dashedName: step-19
+---
+
+# --description--
+
+Finally, return the value of `root` from the `square_root_bisection` method.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1dd722f6e7c8294eeec4.md:10]:
```suggestion
Finally, return the value of `root` from the `square_root_bisection` function.
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1dd722f6e7c8294eeec4.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+
+# --hints--
+
+You should have return statement at the end of the function.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1dd722f6e7c8294eeec4.md:14]:
```suggestion
You should return `root` at the end of the function.
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1e10268f8ec9121446d1.md [hunk 15]
```diff
@@ -0,0 +1,70 @@
+---
+id: 65ef1e10268f8ec9121446d1
+title: Step 20
+challengeType: 20
+dashedName: step-20
+---
+
+# --description--
+
+Outside the function definiton. create a variable `N` and assign the value of `16` to it.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1e10268f8ec9121446d1.md:10]:
```suggestion
Outside the function definiton, create a variable `N` and assign the value of `16` to it.
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1e10268f8ec9121446d1.md [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+
+# --hints--
+
+You should have the variable `N = 16` outside the function definition.
+
+```js
+({ 
+    test: () => 
+    {
+        assert.match(code, /N\s*=\s*16/)
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1e10268f8ec9121446d1.md:20]:
```suggestion
        assert(runPython(`_Node(_code).find_variable("N).is_equivalent("N = 16")`))
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1eac497754cafa12a26c.md [hunk 17]
```diff
@@ -0,0 +1,118 @@
+---
+id: 65ef1eac497754cafa12a26c
+title: Step 21
+challengeType: 20
+dashedName: step-21
+---
+
+# --description--
+
+Call the `square_root_bisection` method with the `N` variable as the argument and print the result.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1eac497754cafa12a26c.md:10]:
```suggestion
Call the `square_root_bisection` function with the `N` variable as the argument and print the result.
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1eac497754cafa12a26c.md [hunk 18]
```diff
@@ -0,0 +1,118 @@
...
+
+Experiment with larger values.
+
+With this, you have successfully implemented the bisection method to find the square root of a number.
+ 
+# --hints--
+
+Call the `square_root_bisection` method with the `N` variable as the argument and print the result.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1eac497754cafa12a26c.md:18]:
```suggestion
You should call the `square_root_bisection` function with the `N` variable as the argument and print the result.
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1eac497754cafa12a26c.md [hunk 19]
```diff
@@ -0,0 +1,118 @@
...
+
+```js
+({ 
+    test: () => 
+    {
+        assert.match(code, /square_root_bisection\(N\)/)
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1eac497754cafa12a26c.md:24]: Reminder: change this once the `find_calls` helper is ready.

**Review by Dario-DC — CHANGES_REQUESTED** [review]
I went trough the steps and left some comments. It's looking good. It'd be nice to have some kind of output before the end (no idea of how to do it, but I'll think about that).

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1eac497754cafa12a26c.md (hunk 19 above)
Comment (zairahira) [reply curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1eac497754cafa12a26c.md]: Done.

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d924ff2e5c7031c6ce8.md (hunk 10 above)
Comment (zairahira) [reply curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d924ff2e5c7031c6ce8.md:10]: I modified the instruction a bit.

File: client/i18n/locales/english/intro.json [hunk 20]
```diff
@@ -852,6 +852,10 @@
         "intro": [
           "This is one of the required projects to claim your certification."
         ]
+      },
+      "learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method": {
+        "title": "Learn Numerical Methods by Finding the Square Root of a Number Using the Bisection Method",
```
[the remark this answers is not in the document]
Comment (fhsinchy) [reply client/i18n/locales/english/intro.json:857]: It is long but I couldn't come with a better name. How about "Learn Numerical Method by Using Bisection to Find the Square Root of a Number"?

**Review by fhsinchy — COMMENTED** [review]
Everything looks fine to me. I have just put forward an alternative title nothing else.

File: client/i18n/locales/english/intro.json (hunk 20 above)
[the remark this answers is not in the document]
Comment (Dario-DC) [reply client/i18n/locales/english/intro.json:857]: I vote for Tom's suggestion: Learn the Bisection Method by Finding the Square Root of a Number

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d924ff2e5c7031c6ce8.md [hunk 21]
```diff
@@ -0,0 +1,77 @@
+---
+id: 65ef1d924ff2e5c7031c6ce8
+title: Step 18
+challengeType: 20
+dashedName: step-18
+---
+
+# --description--
+
+Create an `else` clause to handle the case where the value of `root` is not `None`, indicating that a root hass been found. If it is not `None`, print the message `'The approximate square root of {square_target} is {root}.'`. Remember to format using an f-string.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d924ff2e5c7031c6ce8.md:10]:
```suggestion
Create an `else` clause to handle the case where the value of `root` is not `None`, indicating that a root hass been found. If it is not `None`, print the message `'The square root of {square_target} is approximately {root}'`. Remember to format using an f-string.
```

**Review by Dario-DC — COMMENTED** [review]
Left some suggestions. We should probably start to think where this project is going to be placed.

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1c34a541afc12dbb849d.md [hunk 22]
```diff
@@ -0,0 +1,53 @@
+---
+id: 65ef1c34a541afc12dbb849d
+title: Step 13
+challengeType: 20
+dashedName: step-13
+---
+
+# --description--
+
+The `abs()` function returns the positive value of a number. The returned value, regardless of the sign, is enough to assess if the estimated square root is close enough to the actual value, within a defined tolerance.
```
[the remark this answers is not in the document]
Comment (zairahira) [reply curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1c34a541afc12dbb849d.md:10]: Should the updated instruction be `regardless of the sign` or `regardless of the number sign`?

[the remark this answers is not in the document]
Comment (Dario-DC) [reply curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1c34a541afc12dbb849d.md:10]: My mistake. I edited the suggestion.

File: client/i18n/locales/english/intro.json (hunk 20 above)
[the remark this answers is not in the document]
Comment (zairahira) [reply client/i18n/locales/english/intro.json:857]: Should I also update the dashed names for consistency?

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1c0a03fcabc04ded7e69.md [hunk 23]
```diff
@@ -7,25 +7,35 @@ dashedName: step-12
...
+	values = ["mid = (low + high) / 2", "mid = (high + low) / 2"]
+	any(node.is_equivalent(val) for val in values)
+	`))
         
     }
 
 })
 ```
 
+You should declare a variable square_mid and assign it mid**2 inside the body of the for loop.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1c0a03fcabc04ded7e69.md:33]:
```suggestion
You should declare a variable `square_mid` and assign it `mid**2` inside the body of the for loop.
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1c0a03fcabc04ded7e69.md [hunk 24]
```diff
@@ -0,0 +1,63 @@
...
+        print(f'The square root of {square_target} is 0')
+
+    else:
+        low = 0
+        high = max(1, square_target)
+        root = None
+        
+--fcc-editable-region--
+        for _ in range(max_iterations):
+
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1c0a03fcabc04ded7e69.md:61]:
```suggestion
        for _ in range(max_iterations):
            pass
```

File: client/i18n/locales/english/intro.json (hunk 20 above)
[the remark this answers is not in the document]
Comment (naomi-lgbt) [reply client/i18n/locales/english/intro.json]: Yes

**Review by Dario-DC — APPROVED** [review]
LGTM :+1:  I think we can keep the current order and fix it later when we decide the final order.

File: client/src/pages/learn/scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/index.md [hunk 25]
```diff
@@ -0,0 +1,10 @@
+---
+title: Introduction to the Learn Numerical Methods by Finding the Square Root of a Number Using the Bisection Method
+block: learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method
```
Comment (Dario-DC) [inline client/src/pages/learn/scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/index.md:3]:
```suggestion
title: Introduction to the Learn the Bisection Method by Finding the Square Root of a Number
block: learn-the-bisection-method-by-finding-the-square-root-of-a-number
```

File: client/src/pages/learn/scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/index.md [hunk 26]
```diff
@@ -0,0 +1,10 @@
...
+superBlock: scientific-computing-with-python
+isBeta: true
+---
+
+## Introduction to the Learn Numerical Methods by Finding the Square Root of a Number Using the Bisection Method
```
Comment (Dario-DC) [inline client/src/pages/learn/scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/index.md:8]:
```suggestion
## Introduction to the Learn the Bisection Method by Finding the Square Root of a Number
```

File: curriculum/challenges/_meta/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/meta.json [hunk 27]
```diff
@@ -0,0 +1,101 @@
+{
+  "name": "Learn Numerical Methods by Finding the Square Root of a Number Using the Bisection Method",
```
Comment (Dario-DC) [inline curriculum/challenges/_meta/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/meta.json:2]:
```suggestion
  "name": "Learn the Bisection Method by Finding the Square Root of a Number",
```

File: curriculum/challenges/_meta/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/meta.json [hunk 28]
```diff
@@ -0,0 +1,101 @@
...
+  "isUpcomingChange": true,
+  "usesMultifileEditor": true,
+  "hasEditableBoundaries": true,
+  "dashedName": "learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method",
```
Comment (Dario-DC) [inline curriculum/challenges/_meta/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/meta.json:6]:
```suggestion
  "dashedName": "learn-the-bisection-method-by-finding-the-square-root-of-a-number",
```

**Review by Dario-DC — CHANGES_REQUESTED** [review]
Okay, I missed that. Now it should work

File: client/i18n/locales/english/intro.json [hunk 29]
```diff
@@ -852,6 +852,10 @@
         "intro": [
           "This is one of the required projects to claim your certification."
         ]
+      },
+      "learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method": {
```
Comment (Dario-DC) [inline client/i18n/locales/english/intro.json:856]:
```suggestion
      "learn-the-bisection-method-by-finding-the-square-root-of-a-number": {
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef198fde24dfb7ff675b42.md [hunk 30]
```diff
@@ -0,0 +1,46 @@
+---
+id: 65ef198fde24dfb7ff675b42
+title: Step 4
+challengeType: 20
+dashedName: step-4
+---
+
+# --description--
+
+If the `square_target` is less than `0`, raise a `ValueError` with the message `'Square root of negative number is not defined in real numbers'`. Don't forget to remove the `pass` keyword.
```
Comment (zairahira) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef198fde24dfb7ff675b42.md:10]: Reminder: `valueError` is introduced in the refactored Sudoku solver, might need to update the instruction based on the final order of this project.

File: client/i18n/locales/english/intro.json [hunk 31]
```diff
@@ -852,6 +852,10 @@
         "intro": [
           "This is one of the required projects to claim your certification."
         ]
+      },
+      "learn-the-bisection-method-by-finding-the-square-root-of-a-number": {
+        "title": "Learn the Bisection Method by Finding the Square Root of a Number",
+        "intro": ["", ""]
```
Comment (Dario-DC) [inline client/i18n/locales/english/intro.json:858]: Remember to add intro. We can do it later, after reordering the projects.

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef190c6b51e9b5a5f7ed29.md [hunk 32]
```diff
@@ -0,0 +1,38 @@
...
+challengeType: 20
+dashedName: step-2
+---
+
+# --description--
+
+Give the `square_root_bisection` method the following parameters:
+
+- `square_target`: The number for which you want to find the square root.
+- `tolerance` (optional): The acceptable difference between the square of the approximate root value and the actual target value (default is `1e-7`).
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef190c6b51e9b5a5f7ed29.md:13]: Regarding the tolerance, if we haven't done it yet, maybe we should mention what the notation `1en` means.

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef198fde24dfb7ff675b42.md (hunk 30 above)
Comment (Dario-DC) [reply curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef198fde24dfb7ff675b42.md:10]: Now the `ValueError` is mentioned in the cipher project, too.

**Review by Dario-DC — APPROVED** [review]
Left just a couple of reminders but I think the project is ready :+1:

**Comment by zairahira** [comment]
> Now the ValueError is mentioned in the cipher project, too.

`ValueError` is mentioned in step-65 of the Cipher project, but not introduced in detail. If the Cipher project remains the first one in the certification, it would be the best place to introduce `ValueError`.

**Comment by Dario-DC** [comment]
> > Now the ValueError is mentioned in the cipher project, too.

I was referring to #54452. Anyway, we can add another sentence here to clarify.

**Comment by zairahira** [comment]
> I was referring to #54452. Anyway, we can add another sentence here to clarify.

Sorry, I missed that PR. As the `ValueError` is now introduced in the Cipher project clearly, we can leave updating the Bisection method for now.

**Review by fhsinchy — APPROVED** [review]
I don't have any more suggestion for this PR. I think this is mergeable.

**Pull request merged**